webapp/channels/src/components/youtube_video/youtube_video.tsx

```diff
@@ -46,24 +52,52 @@ export default class YoutubeVideo extends React.PureComponent<Props, State> {
         this.setState({playing: false});
     };
 
+    toggleShortsExpanded = () => {
```

Comment:
```suggestion
toggleShortsExpanded = () => {
    this.setState((prevState) => ({
        shortsExpanded: !prevState.shortsExpanded,
    }));
};
```

best practice: by using the functional pattern for state change in class based components you will always work with the most recent state, and it is essential specially for toggling values, since you can not trust that multiple states changes could be batched/queued. This way you work with the most updated state.